RealmSwift/List.swift

```diff
@@ -508,14 +494,168 @@ extension List: RealmCollection, RangeReplaceableCollection {
     /// :nodoc:
     public func _observe(_ block: @escaping (RealmCollectionChange<AnyRealmCollection<T>>) -> Void) ->
         NotificationToken {
-        let anyCollection = AnyRealmCollection(self)
-        return _rlmArray.addNotificationBlock { _, change, error in
-            block(RealmCollectionChange.fromObjc(value: anyCollection, change: change, error: error))
+            let anyCollection = AnyRealmCollection(self)
+            return _rlmArray.addNotificationBlock { _, change, error in
+                block(RealmCollectionChange.fromObjc(value: anyCollection, change: change, error: error))
+            }
+    }
+}
+
+#if swift(>=4.0)
+// MARK: - MutableCollection conformance, range replaceable collection emulation
+extension List: MutableCollection {
+    public typealias SubSequence = RandomAccessSlice<List>
+
+    /**
+     Returns the objects at the given range (get), or replaces the objects at the
+     given range with new objects (set).
+
+     - warning: Objects may only be set during a write transaction.
+
+     - parameter index: The index of the object to retrieve or replace.
+     */
+    public subscript(bounds: Range<Int>) -> SubSequence {
+        get {
+            return SubSequence(base: self, bounds: bounds)
+        }
+        set {
+            replaceSubrange(bounds.lowerBound..<bounds.upperBound, with: newValue)
         }
     }
+
+    /**
+     Removes the first object in the list. The object is not removed from the
+     Realm that manages it.
+
+     This is a no-op if the list is already empty.
+
+     - warning: This method may only be called during a write transaction.
+     */
+    public func removeFirst() {
+        guard _rlmArray.count > 0 else {
+            return
+        }
+        _rlmArray.removeObject(at: 0)
+    }
+
+    /**
+     Removes the specified number of objects from the beginning of the list. The
+     objects are not removed from the Realm that manages them.
+
+     This is a no-op if the list is already empty.
```

Comment:
This comment is wrong.

Reply:
The current `removeLast()` implementation is a no-op on empty lists because that's what `NSArray` (and thus RLMArray) does, but maybe all of these should be aligned with RRC.